disperser/apiserver/get_blob_status_v2.go

```diff
@@ -13,59 +13,83 @@ import (
 	dispcommon "github.com/Layr-Labs/eigenda/disperser/common"
 	dispv2 "github.com/Layr-Labs/eigenda/disperser/common/v2"
 	blobstore "github.com/Layr-Labs/eigenda/disperser/common/v2/blobstore"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
 func (s *DispersalServerV2) GetBlobStatus(ctx context.Context, req *pb.BlobStatusRequest) (*pb.BlobStatusReply, error) {
+	reply, st := s.getBlobStatus(ctx, req)
+	api.LogResponseStatus(s.logger, st)
+	if st != nil {
+		// nolint:wrapcheck
+		return reply, st.Err()
+	}
+	return reply, nil
+}
+
+func (s *DispersalServerV2) getBlobStatus(
+	ctx context.Context,
+	req *pb.BlobStatusRequest,
+) (*pb.BlobStatusReply, *status.Status) {
 	start := time.Now()
 	defer func() {
 		s.metrics.reportGetBlobStatusLatency(time.Since(start))
 	}()
 
 	if req.GetBlobKey() == nil || len(req.GetBlobKey()) != 32 {
-		return nil, api.NewErrorInvalidArg("blob key must be present and with 32 bytes")
+		return nil, status.New(
+			codes.InvalidArgument,
+			fmt.Sprintf("blob key must be 32 bytes, got %d bytes", len(req.GetBlobKey())),
+		)
 	}
 
 	blobKey, err := corev2.BytesToBlobKey(req.GetBlobKey())
 	if err != nil {
-		return nil, api.NewErrorInvalidArg(fmt.Sprintf("failed to parse the blob key bytes: %v", err))
+		return nil, status.Newf(codes.InvalidArgument, "invalid blob key: %s", req.GetBlobKey())
 	}
 
 	metadata, err := s.blobMetadataStore.GetBlobMetadata(ctx, blobKey)
 	if err != nil {
 		if strings.Contains(err.Error(), "metadata not found") {
 			s.logger.Info("blob metadata not found", "err", err, "blobKey", blobKey.Hex())
-			return nil, api.NewErrorNotFound(fmt.Sprintf("blob metadata not found for blob key: %s", blobKey.Hex()))
+			return nil, status.New(codes.NotFound, "no such blob found")
 		}
 		s.logger.Warn("failed to get blob metadata", "err", err, "blobKey", blobKey.Hex())
-		return nil, api.NewErrorInternal(fmt.Sprintf("failed to get blob metadata: %s", err.Error()))
+		return nil, status.Newf(codes.Internal, "failed to get blob metadata: %v", err)
 	}
 
 	// If the blob is not complete or gathering signatures, return the status without the signed batch
 	if metadata.BlobStatus != dispv2.Complete && metadata.BlobStatus != dispv2.GatheringSignatures {
 		return &pb.BlobStatusReply{
 			Status: metadata.BlobStatus.ToProfobuf(),
-		}, nil
+		}, status.New(codes.OK, "")
 	}
 
 	cert, _, err := s.blobMetadataStore.GetBlobCertificate(ctx, blobKey)
 	if err != nil {
-		s.logger.Error("failed to get blob certificate for blob in GatheringSignatures/Complete status", "err", err, "blobKey", blobKey.Hex())
+		s.logger.Error(
```

Comment:
Any reason why the standard logging above isn't sufficient, and this is logged manually here?

(Same question goes for other manual logs within these utilities)